server/server.go

```diff
@@ -30,52 +35,71 @@ func upload(w http.ResponseWriter, r *http.Request) {
 		arg := r.FormValue(argName)
 		fmt.Println(arg)
 	}
-	defer file.Close()
+	defer func() {
+		err = file.Close()
+		if err != nil {
+			log.Fatal(err)
+			return
+		}
+	}()
 
 	fileHeader := make([]byte,512)
 	if _, err := file.Read(fileHeader)
 		err != nil {
 			log.Fatal(err)
-	}
+		}
 
 	fmt.Printf("Uploaded File: %+v\n", handler.Filename)
```

Comment:
Assuming these print statements are going to be later refactored into HTTP responses.